src/components/RightSidebar/TableOfContents.tsx

```diff
@@ -1,22 +1,27 @@
 import { unescape } from 'html-escaper';
 import type { FunctionalComponent } from 'preact';
-import { useState, useEffect, useRef } from 'preact/hooks';
+import { useState, useEffect } from 'preact/hooks';
 import './TableOfContents.css';
+import type { TocItem } from '../../util/generateToc';
 
 interface Props {
-	headings: { depth: number; slug: string; text: string }[];
+	toc: TocItem[];
 	labels: {
 		onThisPage: string;
 		overview: string;
```

Comment:
Looks like this label is no longer used right because we’re including it in the generated headings? We can remove it here in that case and also stop passing it in the places `<TableOfContents />` gets used.

```suggestion
```